x-pack/plugins/security_solution/server/endpoint/routes/actions/status.test.ts

```diff
@@ -194,360 +220,19 @@ describe('Endpoint Action Status', () => {
-          ],
-          [
-            aMockResponse(actionIds[0], mockAgentId, true),
-            aMockResponse(actionIds[1], mockAgentId, true),
-          ]
-        );
-        (endpointAppContextService.getEndpointMetadataService as jest.Mock) = jest
-          .fn()
-          .mockReturnValue({
-            findHostMetadataForFleetAgents: jest.fn().mockResolvedValue([]),
-          });
-        const response = await getPendingStatus({
-          query: {
-            agent_ids: [mockAgentId],
-          },
-        });
-        expect(response.ok).toBeCalled();
-        expect((response.ok.mock.calls[0][0]?.body as any)?.data).toHaveLength(1);
-        expect((response.ok.mock.calls[0][0]?.body as any)?.data[0].agent_id).toEqual(mockAgentId);
-        expect(
-          (response.ok.mock.calls[0][0]?.body as any)?.data[0].pending_actions.isolate
-        ).toEqual(2);
-      });
-
-      it('should show multiple pending action responses, and their counts', async () => {
-        const mockAgentID = 'XYZABC-000';
-        const actionIds = ['ack_0', 'ack_1', 'ack_2', 'ack_3', 'ack_4'];
-        havingActionsAndResponses(
-          [
-            aMockAction().withAgent(mockAgentID).withAction('isolate').withID(actionIds[0]),
-            aMockAction().withAgent(mockAgentID).withAction('isolate').withID(actionIds[1]),
-            aMockAction().withAgent(mockAgentID).withAction('isolate').withID(actionIds[2]),
-            aMockAction().withAgent(mockAgentID).withAction('unisolate').withID(actionIds[3]),
-            aMockAction().withAgent(mockAgentID).withAction('unisolate').withID(actionIds[4]),
-          ],
-          [
-            aMockResponse(actionIds[0], mockAgentID, true),
-            aMockResponse(actionIds[1], mockAgentID, true),
-            aMockResponse(actionIds[2], mockAgentID, true),
-            aMockResponse(actionIds[3], mockAgentID, true),
-            aMockResponse(actionIds[4], mockAgentID, true),
-          ]
-        );
-        (endpointAppContextService.getEndpointMetadataService as jest.Mock) = jest
-          .fn()
-          .mockReturnValue({
-            findHostMetadataForFleetAgents: jest.fn().mockResolvedValue([]),
-          });
-        const response = await getPendingStatus({
-          query: {
-            agent_ids: [mockAgentID],
-          },
-        });
-        expect(response.ok).toBeCalled();
-        expect((response.ok.mock.calls[0][0]?.body as any)?.data).toHaveLength(1);
-        expect((response.ok.mock.calls[0][0]?.body as any)?.data[0].agent_id).toEqual(mockAgentID);
-        expect(
-          (response.ok.mock.calls[0][0]?.body as any)?.data[0].pending_actions.isolate
-        ).toEqual(3);
-        expect(
-          (response.ok.mock.calls[0][0]?.body as any)?.data[0].pending_actions.unisolate
-        ).toEqual(2);
-      });
-    });
-  });
-
-  describe('response (when pendingActionResponsesWithAck is FALSE)', () => {
```

Comment:
This `describe()` block was testing the same thing as the prior one, but just ensuring that the pending counts was zero. I altered the prior `.describe()` block to include a `.each()` and it now tests both conditions: with `pendingActionResponsesWithAck` set as `true` and `false`. See refactor done at the top of the test file